lib/deref.js

```diff
@@ -247,23 +247,29 @@ module.exports = {
       }
       return { value: 'reference ' + schema.$ref + ' not found in the OpenAPI spec' };
     }
-    if (concreteUtils.compareTypes(schema.type, SCHEMA_TYPES.object) || schema.hasOwnProperty('properties') ||
-      schema.hasOwnProperty('additionalProperties')) {
+
+    const getAdditionalProperties = (schema) => {
+      let additionalProperties;
+      if (_.isBoolean(schema.additionalProperties)) {
+        additionalProperties = schema.additionalProperties;
+      }
+      else {
+        additionalProperties = this.resolveRefs(schema.additionalProperties, parameterSourceOption,
+          components, schemaResolutionCache, resolveFor, resolveTo, stack, _.cloneDeep(seenRef), stackLimit);
```

Comment:
Does this resolve both scenarios:

      additionalProperties:
        type: integer
      additionalProperties:
        $ref:'#/components/schemas/Pet'